include/yaclib/fault/detail/atomic.hpp

```diff
@@ -58,41 +62,52 @@ class AtomicBase : public AtomicWait<Impl, T> {
     return r;
   }
 
-  // TODO(myannyax) Add probability of random fail for compare_exchange_weak
   bool compare_exchange_weak(T& expected, T desired, std::memory_order success, std::memory_order failure) noexcept {
+    if (ShouldFailAtomicWeak()) {
```

Comment:
I think will be better:
FaultInject();
if (...) false
op
FaultInject()

Reply:
Also maybe if should fail we should read current value to expected, what do you think?